### ISSUE TITLE
Bug: Multi-line strings + chaining causes indentation error

### ISSUE THREAD

**Issue by alubbe** [clutchski/coffeelint#332]
This code causes an identation error on the last line, even though it shouldn't:
```coffee
xmlrpc.createClient
  host: "XX"
  path: "YY?a=#{b}&\
         c=#{f x, y}"
.methodCall 'getPosts', [bbb: 12345]
```

**Comment by lirumlarum** [comment]
at least have an option to enable that the linter ignore validation of multiline strings with different indention, would be nice. Same issue with functions, sometimes the line is getting to long and it would help with coding style to have the possibility to disable linting, the following is not valid and leads to an error.

```javascript
 service = ($state, $http, Chart, $filter, $currentAsset, $plugin, 
   $formatter, $dateRange, $translate, $chartTooltip
 ) ->
```

**Comment by AsaAyers** [comment]
The best way to get that option is for someone to try implementing it. You may have a good idea there with some kind of option to make it less strict, but I have other higher priorities so I can say that I won't be addressing this any time soon.

One thing I thought about doing is adding an option `{ "indentation": { "mode": "brain-dead", "level": 2 } }`. In that idea of a mode we totally ignore everything about the language and just make sure the indentation is a multiple of `level`. Just about anything that compiles would be considered valid under that mode.

**Comment by lirumlarum** [comment]
agree so far, but what do you say about the second scenario, its obviously a bug isn't it? The code is following right indenting but is invalid because the following fluent calls are not align with the angular variable.

**Comment by AsaAyers** [comment]
In the 2 years I've been managing this project there has never been a time when everyone was happy with indentation. There have been many attempts to fix it but at this point I don't believe we're ever going to make everyone happy.

Whether it's a "bug" is very subjective. I'm not going to call it a bug, it's a style difference and the style you have chosen is just very difficult to implement in CoffeeLint.

**Comment by lirumlarum** [comment]
maybe, but i would expect a linter to lint in my coding style not in the coding style that is easy to implement. I mean when the code is valid and is following the same style like indention level consistent the linter should mark that as valid. Don't you think that the linter should provide more config options in that topic then? I mean the code is valid for sure and is following the right and consistent indention level thats why the linter should be able to handle that.

**Comment by AsaAyers** [comment]
so what I'm hearing is "Because my code compiles it should pass the linter". The whole point of the linter is to warn you about code that compiles, but you may want to change anyway.

**Comment by AsaAyers** [comment]
> i would expect a linter to lint in my coding style not in the coding style that is easy to implement. 

When your pull request is ready it will get top priority and will get merged and released ASAP.

**Comment by AsaAyers** [comment]
Also, let's get one thing straight here.  There is nothing "easy" about linting this. We have spent MANY hours making the linter work the way it does. The indentation is by far the most complex rule. It's 168 lines where the next is 85 lines for `braces_spacing`. @swang did an amazing job at improving the indentation last year among other things.

**Comment by lirumlarum** [comment]
it wasn't my intention to blame your work guys. The linting just don't work as expected for me thats all. I think it's a complex topic, what i meant is: the current implementation is less complex as i would need it. Easy implementation is the wrong term i guess. I'm pretty busy right now but hopefully i can give it try soon to improve it! thanks for your feedback!

**Comment by swang** [comment]
If you would like indentation fixed, please encourage the coffeescript team to fix the lexer. They rewrite the code for their own style reasons before tokenzing which throws off generating proper INDENT tokens that can be read.

**Comment by swang** [comment]
This has been self-resolved with #492, which seems to now ignore indentation form strings in object properties.
